examples/go/vm-reconfig-via-tag/handler/handler.go

```diff
@@ -0,0 +1,564 @@
+		}
+
+		if err == nil {
+			tagObjs, err = vsClt.attachedTags(ctx, *vmMOR)
+		}
+	}
+
+	if err != nil {
+		return handlerResponseWithError(
+			"get tags attached to VM: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// Check SOAP connection by getting managed object VM.
+	moVM, err := vsClt.moVirtualMachine(ctx, *vmMOR)
+	if err != nil {
+		// Retry once after creating new client.
+		if strings.Contains(err.Error(), "NotAuthenticated") {
+			if debug {
+				log.Println("NotAuthenticated. Retry connection to VAPI.")
+			}
+
+			vsClt, err = newClient(ctx, cfg)
+		}
+
+		if err == nil {
+			moVM, err = vsClt.moVirtualMachine(ctx, *vmMOR)
+		}
+	}
+
+	if err != nil {
+		return handlerResponseWithError(
+			"retrieve current VM configs: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	// Get the current configurations in a list from the VM object.
+	currCfgs := currentConfigs(ctx, moVM)
+
+	// Look for configurations that need to be applied to the VM.
+	newCfgs, err := vsClt.unappliedConfigs(ctx, tagObjs, currCfgs)
+	if err != nil {
+		return handlerResponseWithError(
+			"select only config tags: %w",
+			http.StatusInternalServerError,
+			err,
+		)
+	}
+
+	var task *object.Task
+
+	if len(newCfgs) > 0 {
+		task, err = vsClt.applyCfgs(ctx, moVM, *vmMOR, newCfgs)
+		if err != nil {
+			return handlerResponseWithError(
+				"apply config tags: %w",
+				http.StatusInternalServerError,
+				err,
+			)
+		}
+	}
+
+	message := appliedConfigsMessage(task, newCfgs)
+
+	return handler.Response{
+		Body:       []byte(message),
+		StatusCode: http.StatusOK,
+	}, nil
+}
+
+func setVerbosity() bool {
+	return os.Getenv("write_debug") == "true"
+}
+
+func loadConfig(path string) (*vcConfig, error) {
+	var cfg vcConfig
+
+	secret, err := toml.LoadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("load vcconfig.toml: %w", err)
+	}
+
+	err = secret.Unmarshal(&cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal vcconfig.toml: %w", err)
+	}
+
+	err = validateConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("validate vcconfig: %w", err)
+	}
+
+	if debug {
+		log.Println("vcconfig.toml loaded, unmarshalled, and validated.")
+	}
+
+	return &cfg, nil
+}
+
+func handlerResponseWithError(msg string, statusCode int, err error) (handler.Response, error) {
+	wrapErr := fmt.Errorf(msg, err)
+	log.Println(wrapErr.Error())
+
+	return handler.Response{
+		Body:       []byte(wrapErr.Error()),
+		StatusCode: statusCode,
+	}, wrapErr
+}
+
+// ValidateConfig ensures the bare minimum of information is in the config file.
+func validateConfig(cfg vcConfig) error {
+	reqFields := map[string]string{
+		"vcenter server":   cfg.VCenter.Server,
+		"vcenter user":     cfg.VCenter.User,
+		"vcenter password": cfg.VCenter.Password,
+	}
+
+	// Multiple fields may be missing, but err on the first encountered.
+	for k, v := range reqFields {
+		if v == "" {
+			return errors.New("required field(s) missing, including " + k)
```

Comment:
Can I suggest the following (more precise) error message to ease debugging?

```suggestion
			return errors.New("required field missing: %q" + k)
```